.travis.yml

```diff
@@ -36,6 +37,16 @@ matrix:
       env: TASK=gpu METHOD=pip
     - os: osx
       env: TASK=pylint
+    - os: osx
+      env: TASK=if-else
+    - os: osx
+      env: TASK=proto
+    - os: osx
+      env: TASK=regular PYTHON_VERSION=2.7
```

Comment:
please see https://github.com/Microsoft/LightGBM/pull/904#issuecomment-328732939

Reply:
@henry0312 I saw. Do you have an agreement yet? I can add it later, I think it's a waste of resource to run those python tests I didn't alter multiple time now.

Reply:
> Do you have an agreement yet?

No, I don't know if @guolinke aggrees with it yet, but I think we should run tests with Python 2.7.x and Python 3.6.x at least.

> I think it's a waste of resource to run those python tests I didn't alter multiple time now.

I got it!
Yeah, Travis always faces with lacks of instances of OSX.
We have no choice but to reduce test cases for OSX 😢